syncmaster/settings/__init__.py

```diff
@@ -1,78 +1,71 @@
 
-from pydantic import model_validator
+from pydantic import Field, model_validator
 from pydantic.types import ImportString
 from pydantic_settings import BaseSettings
 
-if TYPE_CHECKING:
-    pass
+from syncmaster.settings.broker import RabbitMQSettings
+from syncmaster.settings.database import DatabaseSettings
+from syncmaster.settings.server import ServerSettings
 
 
 class EnvTypes(StrEnum):
     LOCAL = "LOCAL"
 
 
 class Settings(BaseSettings):
-    ENV: EnvTypes = EnvTypes.LOCAL
-    DEBUG: bool = True
+    """Syncmaster backend settings.
+
+    Backend can be configured in 2 ways:
+
+    * By explicitly passing ``settings`` object as an argument to :obj:`application_factory
+     <syncmaster.backend.main.application_factory>`
+    * By setting up environment variables matching a specific key.
+
+        All environment variable names are written in uppercase and should be prefixed with ``SYNCMASTER__``.
+        Nested items are delimited with ``__``.
+
+
+    More details can be found in
+    `Pydantic documentation <https://docs.pydantic.dev/latest/concepts/pydantic_settings/>`_.
+
+    Examples
+    --------
+
+    .. code-block:: bash
+
+        # same as settings.database.url = "postgresql+asyncpg://postgres:postgres@localhost:5432/syncmaster"
+        SYNCMASTER__DATABASE__URL=postgresql+asyncpg://postgres:postgres@localhost:5432/syncmaster
+
+        # same as settings.server.debug = True
+        SYNCMASTER__SERVER__DEBUG=True
+    """
 
     PROJECT_NAME: str = "SyncMaster"
 
     SECRET_KEY: str = "secret"
     SECURITY_ALGORITHM: str = "HS256"
     CRYPTO_KEY: str = "UBgPTioFrtH2unlC4XFDiGf5sYfzbdSf_VgiUSaQc94="
 
-    POSTGRES_HOST: str
-    POSTGRES_PORT: int
-    POSTGRES_DB: str
-    POSTGRES_USER: str
-    POSTGRES_PASSWORD: str
-
-    RABBITMQ_HOST: str
-    RABBITMQ_PORT: int
-    RABBITMQ_USER: str
-    RABBITMQ_PASSWORD: str
-
-    def build_db_connection_uri(
-        self,
-        *,
-        driver: str | None = None,
-        host: str | None = None,
-        port: int | None = None,
-        user: str | None = None,
-        password: str | None = None,
-        database: str | None = None,
-    ) -> str:
-        return "postgresql+{}://{}:{}@{}:{}/{}".format(
-            driver or "asyncpg",
-            user or self.POSTGRES_USER,
-            password or self.POSTGRES_PASSWORD,
-            host or self.POSTGRES_HOST,
-            port or self.POSTGRES_PORT,
-            database or self.POSTGRES_DB,
-        )
-
-    def build_rabbit_connection_uri(
-        self,
-        *,
-        host: str | None = None,
-        port: int | None = None,
-        user: str | None = None,
-        password: str | None = None,
-    ) -> str:
-        return "amqp://{}:{}@{}:{}//".format(
-            user or self.RABBITMQ_USER,
-            password or self.RABBITMQ_PASSWORD,
-            host or self.RABBITMQ_HOST,
-            port or self.RABBITMQ_PORT,
-        )
+    LOG_URL_TEMPLATE: str = ""
```

Comment:
IMHO log_url template should be implemented in a separated MR

Reply:
removed from current PR